### ISSUE TITLE
Build OCI artifacts and save to the registry

### ISSUE THREAD

**Issue by nabokihms** [werf/werf#5995]
### Before proceeding

- [X] I didn't find a similar [issue](https://github.com/werf/werf/issues)

### Problem

Currently, there is no functionality in werf to save images as OCI artifacts during the build stage.
Some artifacts are required to be saved in the container registry for further distribution, e.g., to build and distribute binaries.

### Solution (if you have one)

Introduce the capability to save images as OCI artifacts during the build process in werf. This feature would allow users to produce OCI-compliant artifacts directly from their builds, enhancing interoperability and compatibility with other container tools and platforms.

The example of the possible werf.yaml to show the idea.

```yaml
image: my-app
artifact: true
git:
- add: /
  to: /src
shell:
  install:
  - go build /src/main.go -o /my-app
---
image: my-app-binary
oci: true
fromArtifact: my-app
filePath: /my-app
mediatype: application/vnd.my-company.golang.binary.content.v1
```

As a result, the `/my-app` binary from the `my-app` artifact will be uploaded to the container registry.

### Additional information

By enabling the option to save images as OCI artifacts, werf users can benefit from improved portability and standardized image distribution. OCI artifacts offer advantages such as reduced complexity, increased security, and better support for multi-platform environments. This enhancement aligns with industry best practices and enhances the overall usability and versatility of the werf tool.

For more information about OCI (Open Container Initiative) standards, please refer to the official OCI website: [Open Container Initiative](https://opencontainers.org/).

**Comment by alexey-igrychev** [comment]
No work has been done on this and there's been no further demand since it was filed, so we're closing it as not planned. If you have a concrete use case that OCI artifacts would unblock, please open a new issue describing it — that's much easier to prioritise.
